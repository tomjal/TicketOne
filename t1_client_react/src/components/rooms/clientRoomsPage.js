import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import {
  createClientRoom,
  markRoomAsResolved,
  markRoomAsUnresolved
} from './../../actions/roomsActionCreators';
import { sendMessageToRoom } from './../../actions/messagesActionCreators';
import { getClientInitRoomsData } from './../../actions/combinedActionCreators';

import RoomWidget from './widgets/roomWidget';
import CreateNewRoomWidget from './widgets/createNewRoomWidget';

const mapStateToProps = (state) => {
  return {
    id: state.context.id,
    role: state.context.role,
    messages: state.messages,
    rooms: state.rooms
  }
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    getClientInitRoomsData,
    createClientRoom,
    sendMessageToRoom,
    markRoomAsResolved,
    markRoomAsUnresolved,
  }, dispatch);
};

export class ClientRoomsPage extends Component {
  componentDidMount() {
    this.props.getClientInitRoomsData(this.props.id);
  }
  sendMessageCallback = (message, roomId) => {
    this.props.sendMessageToRoom(message, roomId, this.props.role, this.props.id);
  }
  createRoomCallback = (roomTopic) => {
    this.props.createClientRoom(this.props.id, roomTopic);
  }
  render() {
    const { id, role, messages, rooms } = this.props;
    return (
      <div className="room-flex-container">
        {rooms.map((room, i) => {
          let localMessages = [];
          if (messages[room.id]) {
            localMessages = messages[room.id];
          }
          return <RoomWidget key={i}
            id={room.id}
            topic={room.roomTopic}
            isOpened={room.isOpened}
            isSolved={room.isSolved}
            clientId={id}
            role={role}
            sendMessageCallback={this.sendMessageCallback}
            markAsResolvedCallback={this.props.markRoomAsResolved}
            markAsUnresolvedCallback={this.props.markRoomAsUnresolved}
            messages={localMessages} />;
        })}
        <CreateNewRoomWidget
          createRoomCallback={this.createRoomCallback} />
      </div>
    );
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ClientRoomsPage);
