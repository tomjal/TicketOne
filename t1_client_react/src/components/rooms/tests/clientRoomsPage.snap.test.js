import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { ClientRoomsPage } from './../clientRoomsPage';

test('ClientRoomsPage snapshot', () => {
  const props = {
    id: 0,
    role: "",
    messages: [],
    rooms: [],
    getClientInitRoomsData: jest.fn(),
    createClientRoom: jest.fn(),
    sendMessageToRoom: jest.fn(),
    markRoomAsResolved: jest.fn(),
    markRoomAsUnresolved: jest.fn()
  }

  const wrap = shallow(<ClientRoomsPage {...props} />);

  expect(shallowToJson(wrap)).toMatchSnapshot();
});