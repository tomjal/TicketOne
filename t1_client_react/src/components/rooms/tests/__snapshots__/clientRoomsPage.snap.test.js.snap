// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientRoomsPage snapshot 1`] = `
<div
  className="room-flex-container"
>
  <CreateNewRoomWidget
    createRoomCallback={[Function]}
  />
</div>
`;
